addons/mail/static/src/models/message/message.js

```diff
@@ -256,6 +258,22 @@ function factory(dependencies) {
             }));
         }
 
+        /**
+         * Edits the message's body.
+         * 
+         * @param {String} body the new body of the message
+         */
+        async edit(body) {
+            const payload = await this.async(() => this.env.services.rpc({
```

Comment:
`this.async` is actually useless, a mistake from the past.

Just ensure that whatever you are trying to do after await won't crash, eg. in this case return directly `this.messaging` is no longer set.